modules/avcodec/encode.c

```diff
@@ -205,8 +210,16 @@ static int open_encoder(struct videnc_state *st,
 	av_opt_set_defaults(st->ctx);
 
 	st->ctx->bit_rate  = prm->bitrate;
-	st->ctx->width     = size->w;
-	st->ctx->height    = size->h;
+
+	if (sz_dst.w <= 0) {;
+		sz_dst.w = size->w;
+	}
+	if (sz_dst.h <= 0) {
+		sz_dst.h = size->h;
+	}
+
+	st->ctx->width     = sz_dst.w;
+	st->ctx->height    = sz_dst.h;
```

Comment:
During preparing the patch I didn't check the details of avcodec if it will be required or not. This code part was only updated to use sz_dst instead of size->w, as used before.

Having a quick look at the FFmpeg documentation it must be set during the encoding process:
```
int AVCodecContext::width
- encoding: MUST be set by user.
- decoding: May be set by the user before opening the decoder if known e.g. from the container. Some decoders will require the dimensions to be set by the caller. During decoding, the decoder may overwrite those values as required while parsing the data.
```
https://ffmpeg.org/doxygen/3.2/structAVCodecContext.html#a0d8f46461754e8abea0847dcbc41b956